libfoundation/src/foundation-string.cpp

```diff
@@ -4983,9 +4987,60 @@ bool MCStringConvertToSysString(MCStringRef p_string, const char * &r_system_str
     t_sys_string[t_sys_len] = '\0';
 
 	r_system_string = t_sys_string;
+    r_byte_count = t_sys_len;
+    
```

Comment:
The NUL byte is implicit (i.e. r_byte_count should be the number of bytes encoded not including NUL terminator - which I believe it is).